skbio/io/format/tests/test_stockholm.py

```diff
@@ -376,5 +401,301 @@ def test_handles_missing_metadata_efficiently(self):
+                              metadata=OrderedDict([('DT', 'February 1, 2016'),
+                                                    ('NM', 'Unknown')])),
+                          DNA('ACCTGAGCCACAGTAGAAGT'),
+                          DNA('CCCTTCGCTGGAAATGTATG',
+                              metadata={'DT': 'Unknown'},
+                              positional_metadata=OrderedDict([('AS',
+                                                                list('CCGAAAGT'
+                                                                     'CGTTCGA'
+                                                                     'AAATG')),
+                                                               ('SS',
+                                                                list('GGCGAGTC'
+                                                                     'GTTCGAGC'
+                                                                     'TGG'
+                                                                     'C'))]))],
+                         metadata=OrderedDict([('NM', 'Kestrel Gorlick'),
+                                              ('DT', 'February 11, 2016'),
+                                              ('FN', 'Writer test file')]),
+                         positional_metadata=OrderedDict([('AS_cons',
+                                                           list('CGTTCGTTCTAAC'
+                                                                'AATTCCA')),
+                                                          ('SS_cons',
+                                                           list('GGCGCTACGACCT'
+                                                                'ACGACCG'))]),
+                         index=['seq1', 'seq2', 'seq3'])
+        fh = io.StringIO()
+        _tabular_msa_to_stockholm(msa, fh)
+        obs = fh.getvalue()
+        fh.close()
+        with io.open(fp) as fh:
+            exp = fh.read()
+        self.assertEqual(obs, exp)
+
+    def test_msa_to_stockholm_minimal(self):
+        fp = get_data_path('stockholm_minimal')
+        msa = TabularMSA([DNA('TGTGTCGCAGTTGTCGTTTG')], index=['0235244'])
+        fh = io.StringIO()
+        _tabular_msa_to_stockholm(msa, fh)
+        obs = fh.getvalue()
+        fh.close()
+        with io.open(fp) as fh:
+            exp = fh.read()
+        self.assertEqual(obs, exp)
+
+    def test_msa_to_stockholm_single_tree(self):
+        fp = get_data_path('stockholm_single_tree_without_id')
+        msa = TabularMSA([], metadata=OrderedDict([('NH', 'ABCD')]))
+        fh = io.StringIO()
+        _tabular_msa_to_stockholm(msa, fh)
+        obs = fh.getvalue()
+        fh.close()
+        with io.open(fp) as fh:
+            exp = fh.read()
+        self.assertEqual(obs, exp)
+
+    def test_msa_to_stockholm_single_tree_as_dict(self):
+        fp = get_data_path('stockholm_single_tree_without_id')
+        msa = TabularMSA([], metadata={'NH': 'ABCD'})
```

Comment:
The test I was asking for should test that this works as expected (the inner dict is the important part, older versions of the code only worked when it was an `OrderedDict`):

```python
metadata={'NH': {'tree-id': 'tree data'}}
```